### PR TITLE
Delete old console and event listeners and event-listeners

### PR DESCRIPTION
**Pull request by bvaughn**
Builds on top of #7641 and deletes additional things:
- [x] Event types state
  - I'm not 100% about this one because the `Matches` UI component was referencing this Redux node, _but_ I'm not sure it was ever being used. (I didn't see anything actually setting the `enableEventLink` setting to true– so maybe it was just an old orphaned thing?)
- [x] Analysis logic in `src/ui/actions/logpoint`
   - This overlaps with the logic in the new Console. We don't need to run extra analysis. We should just share data between the old and new architecture.
- [x] Fixed a couple of bugs (see https://github.com/replayio/devtools/pull/7643#discussion_r969826546, https://github.com/replayio/devtools/pull/7643#pullrequestreview-1106084427, https://github.com/replayio/devtools/pull/7643#discussion_r969822136)

### CODE REVIEW

**Comment by markerikson** [comment]
Per Jaril:

> Matches was a (very) hacky implementation of connecting a user event (eg mouse click) to the click handler
> Idea is if there was a click in the code, you'd want to be one click away from the react onclick callback
> but it was difficult to pull off for many reasons 🥲
> something something react handled events in a way that made it complicated to just get the location for that callback (i think)
> and then how that's handled was different across react versions

**Review by markerikson — APPROVED** [review]
I am in favor of going ahead and merging this PR instead of #7641 .

Reviewed both, they seem good to me.

LET'S KILL THAT CONSOLE!

**Comment by bvaughn** [comment]
PR #7651 will help unblock this.

File: packages/bvaughn-architecture-demo/src/suspense/PointsCache.ts [hunk 1]
```diff
@@ -73,6 +98,31 @@ export function getHitPointsForLocation(
...
+  condition: string | null,
+  focusRange: TimeStampedPointRange | null
+): Promise<HitPointsAndStatusTuple> {
+  try {
+    return getHitPointsForLocation(client, location, condition, focusRange);
+  } catch (errorOrPromise) {
+    if (
+      errorOrPromise != null &&
+      typeof errorOrPromise === "object" &&
+      typeof errorOrPromise.hasOwnProperty("then")
```
Comment (eddyw) [inline packages/bvaughn-architecture-demo/src/suspense/PointsCache.ts:116]: 👀 the condition will be true with any object even when there is no `then` own property because `typeof` operator always returns a string
```suggestion
      errorOrPromise.hasOwnProperty("then")
```

Comment (bvaughn) [reply packages/bvaughn-architecture-demo/src/suspense/PointsCache.ts:116]: Nice catch! Thanks :)

File: src/devtools/client/debugger/src/components/Editor/Breakpoints/Panel/Panel.tsx [hunk 2]
```diff
@@ -53,35 +48,71 @@ const connector = connect(
...
+  const filteredHitPoints = useMemo(() => {
+    return hitPoints
+      ? hitPoints.filter(hitPoint => {
+          return (
+            hitPoint.time >= unsafeFocusRegionForSuspense.beginTime &&
+            hitPoint.time <= unsafeFocusRegionForSuspense.endTime
+          );
+        })
+      : null;
+  }, [hitPoints, unsafeFocusRegionForSuspense]);
```
Comment (bvaughn) [inline src/devtools/client/debugger/src/components/Editor/Breakpoints/Panel/Panel.tsx:107]: This section re-uses our new Suspense caches to load hit points data. Then we have a secondary pass to filter the hit points by the user-visible focus region. This is necessary, unfortunately, because we often ask for a focus region that's bigger than the time the user requested.

File: src/ui/components/PrefixBadge.tsx [hunk 3]
```diff
@@ -15,7 +15,6 @@ export default function PrefixBadgeButton({ breakpoint }: { breakpoint: Breakpoi
 
   return (
     <PrefixBadgePicker
-      key={breakpoint.options.prefixBadge}
```
Comment (bvaughn) [inline src/ui/components/PrefixBadge.tsx:18]: I'm not sure why we were doing this before, but it was breaking the prefix badge close animation. Now it works again.

File: src/ui/components/Timeline/UnloadedRegions.tsx [hunk 4]
```diff
@@ -31,7 +27,21 @@ export const UnloadedRegions: FC = () => {
...
+    const userVisibleTimeStampedPointRange: TimeStampedPointRange = {
+      begin: {
+        point: unsafe.begin.point,
+        time: unsafe.beginTime,
+      },
+      end: {
+        point: unsafe.end.point,
+        time: unsafe.endTime,
+      },
+    };
```
Comment (bvaughn) [inline src/ui/components/Timeline/UnloadedRegions.tsx:42]: This fixes an issue on the `main` branch where our mini-timeline showed a _larger_ loaded region than the user had focused on, which was confusing.

Before:
Screen Shot 2022-09-13 at 12 08 21 PM

After:
Screen Shot 2022-09-13 at 12 08 42 PM

**Comment by bvaughn** [comment]
I think this is ready for a final review 😄

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimeline.tsx [hunk 5]
```diff
@@ -94,20 +91,22 @@ function BreakpointTimeline({
...
     const time = Math.ceil(beginTime + (endTime - beginTime) * clickPosition);
     setHoveredTime(time);
   };
-  const onMouseLeave = (e: React.MouseEvent) => setHoveredTime(null);
+  const onMouseLeave = () => setHoveredTime(null);
 
   const hoverPercent = `${getVisiblePosition({ time: hoveredTime, zoom: zoomRegion }) * 100}%`;
   const percent = getVisiblePosition({ time: currentTime, zoom: zoomRegion }) * 100;
 
+  // TODO [bvaughn] UnloadedRegions should be loaded + focused
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimeline.tsx:108]: is this todo okay?

File: src/ui/actions/find-tests.ts [hunk 6]
```diff
@@ -73,7 +71,6 @@ return [{
 // Mapper extracting the "name" and "message" properties from a thrown exception
 // for use in test failure messages.
 const JestExceptionMapper = `
-${Helpers}
```
Comment (jasonLaster) [inline src/ui/actions/find-tests.ts:76]: why is it okay to remove these? Seems unrelated

File: src/ui/components/PrefixBadge.tsx (hunk 3 above)
Comment (jasonLaster) [reply src/ui/components/PrefixBadge.tsx:18]: odd

File: src/ui/actions/find-tests.ts (hunk 6 above)
Comment (markerikson) [reply src/ui/actions/find-tests.ts:76]: Good catch - the `Helpers` string was:

```ts
export const Helpers = `
  const finalData = { frames: [], scopes: [], objects: [] };
  function addPauseData({ frames, scopes, objects }) {
    finalData.frames.push(...(frames || []));
    finalData.scopes.push(...(scopes || []));
    finalData.objects.push(...(objects || []));
  }
  function getTopFrame() {
    const { frame, data } = sendCommand("Pause.getTopFrame");
    addPauseData(data);
    return finalData.frames.find(f => f.frameId == frame);
  }
`;
```

Note that it adds `const finalData`.

I see a reference to `finalData` in one of the Jest test-finding mappers:

```ts
const JestExceptionMapper = `
${Helpers}
const { point, time, pauseId } = input;
const { frameId, location } = getTopFrame();
const { exception, data: exceptionData } = sendCommand("Pause.getExceptionValue");
addPauseData(exceptionData);
return [{
  key: point,
  value: { time, pauseId, location, exception, data: finalData },
}];
`;
```

So we should re-add `Helpers` somewhere, and restore those lines here.

File: src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimeline.tsx (hunk 5 above)
Comment (bvaughn) [reply src/devtools/client/debugger/src/components/SecondaryPanes/Breakpoints/BreakpointTimeline.tsx:108]: Ooops. Forgot to delete this comment. Yeah, I already did that. It's what was fixed here: https://github.com/replayio/devtools/pull/7643/files/606ba4950b4c24a83cf748ef3af266aa06b71afd#diff-c9da310bb17d439da080e71962563ba5a48b4740dd69f5243d3f8c33fff2f8d7

File: src/ui/actions/find-tests.ts (hunk 6 above)
Comment (bvaughn) [reply src/ui/actions/find-tests.ts:76]: It's related. This is in `src/ui/actions` so it's Redux stuff, which this PR is deleting a bunch of. In this case, "Helpers" was referring to an export from `actions/logpoint` which was deleted in this PR :)

Comment (markerikson) [reply src/ui/actions/find-tests.ts:76]: Yeah, what I'm seeing is that the `JestExceptionMapper` string requires one of the variables created by that `Helper` string, so we need to bring it back.

**Comment by bvaughn** [comment]
Cool. Good catch. Thanks.

Fixed that issue and rebased. I think we should be unblocked to land today, assuming CI is happy.

**Comment by bvaughn** [comment]
The failing Playwright test is a false negative, and has been fixed by #7744. Otherwise things look green. I think we're good to land this.
